lib/Serialization/Deserialization.cpp

```diff
@@ -166,10 +167,27 @@ static void skipRecord(llvm::BitstreamCursor &cursor, unsigned recordKind) {
 }
 
 void ModuleFile::fatal(llvm::Error error) const {
+  Core->fatal(diagnoseFatal(std::move(error)));
+}
+
+llvm::Error ModuleFile::diagnoseFatal(llvm::Error error) const {
   if (FileContext)
     getContext().Diags.diagnose(SourceLoc(), diag::serialization_fatal,
                                 Core->Name);
-  Core->fatal(std::move(error));
+  std::string msg;
+  {
+    llvm::raw_string_ostream os(msg);
+    Core->outputDiagnosticInfo(os);
+  }
+  msg += ": " + toString(std::move(error));
+  llvm::dbgs() << "*** DESERIALIZATION FAILURE ***\n";
```

Comment:
Why choose `llvm::dbgs` here? IIUC, in theory we don't have any of these errors when things are working well so we might as well keep it on stdout for investigation.

Does it help keeping the error in the LLDB logs?

Reply:
This is no longer relevant.